src/backend_api.cpp

```diff
@@ -0,0 +1,268 @@
+#include <miopen/backend_api.hpp>
+#include <miopen/miopen.h>
+#include <miopen/errors.hpp>
+#include <new>
+
+extern "C" miopenStatus_t
+miopenBackendCreateDescriptor(miopenBackendDescriptorType_t descriptorType,
+                              miopenBackendDescriptor_t* descriptor)
+{
+    switch(descriptorType)
+    {
+    case MIOPEN_BACKEND_TENSOR_DESCRIPTOR:
+        // TODO: add logging
+        // TODO: get rid of unnecessary throws
+        return miopen::try_(
+            [&] {
+                miopen::backend_api::BackendTensorDescriptor* newDescriptor =
+                    new(std::nothrow) miopen::backend_api::BackendTensorDescriptor();
+
+                if(!newDescriptor)
+                {
+                    throw miopen::Exception(miopenStatusAllocFailed);
+                }
+
+                try
```

Comment:
why is this needed? we should not use `try/catch` as that's what `miopen::try_` is for.